File: annotation/src/main/java/com/github/snowdream/toybricks/annotation/Implementation.java
package com.github.snowdream.toybricks.annotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Created by snowdream on 17/2/10.
 */
@Retention(RetentionPolicy.CLASS)
@Target({ElementType.TYPE})
public @interface Implementation {
     Class value();

     boolean global() default false;

     boolean singleton() default false;
}

File: annotation/src/main/kotlin/com/github/snowdream/toybricks/annotation/Implementation.kt
package com.github.snowdream.toybricks.annotation

import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy
import kotlin.reflect.KClass

/**
 * Created by snowdream on 17/2/10.
 */
@Retention(RetentionPolicy.CLASS)
@Target(AnnotationTarget.CLASS, AnnotationTarget.FILE)
annotation class Implementation(val value: KClass<*>, val global: Boolean = false, val singleton: Boolean = false)
